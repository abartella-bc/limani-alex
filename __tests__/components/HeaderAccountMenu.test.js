/*
Copyright 2023 BlueCat Networks Inc.

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.
*/

import { shallow } from 'enzyme';
import HeaderAccountMenu from '../../src/header/HeaderAccountMenu';
import setLanguage from '../../src/functions/setLanguage';

jest.unmock('../../src/header/HeaderAccountMenu');

jest.mock('../../src/hooks/usePlatformData', () =>
    jest.fn(() => {
        return {
            data: {
                user: {
                    userName: 'varUsername',
                    userType: 'varUserType',
                },
            },
        };
    }),
);

describe('HeaderAccountMenu', () => {
    describe('Rendering', () => {
        it('Render HeaderAccountMenu component with default props', () => {
            const wrapper = shallow(<HeaderAccountMenu />);
            expect(wrapper.getElement()).toMatchSnapshot();
        });

        it('Render HeaderAccountMenu component with props ', () => {
            const wrapper = shallow(
                <HeaderAccountMenu className='varClassName' />,
            );
            expect(wrapper.getElement()).toMatchSnapshot();
        });

        it('Render HeaderAccountMenu component with pseudo translated text', () => {
            setLanguage('zz').finally(() => {
                const wrapper = shallow(<HeaderAccountMenu />);
                expect(wrapper.getElement()).toMatchSnapshot();
            });
        });
    });
});
