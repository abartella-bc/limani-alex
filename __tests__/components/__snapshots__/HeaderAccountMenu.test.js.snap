// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderAccountMenu Rendering Render HeaderAccountMenu component with default props 1`] = `
<div
  className="HeaderAccountMenu"
>
  <button
    aria-controls={null}
    aria-expanded={false}
    aria-haspopup="true"
    className="HeaderAccountMenu__button"
    id="accountMenuBtn"
    onClick={[Function]}
    onKeyDown={[Function]}
    type="button"
  >
    <HeaderIcon
      icon={
        Object {
          "icon": Array [
            448,
            512,
            Array [
              128100,
              62144,
            ],
            "f007",
            "M224 256A128 128 0 1 0 224 0a128 128 0 1 0 0 256zm-45.7 48C79.8 304 0 383.8 0 482.3C0 498.7 13.3 512 29.7 512H418.3c16.4 0 29.7-13.3 29.7-29.7C448 383.8 368.2 304 269.7 304H178.3z",
          ],
          "iconName": "user",
          "prefix": "fas",
        }
      }
      label="Account"
    />
  </button>
</div>
`;

exports[`HeaderAccountMenu Rendering Render HeaderAccountMenu component with props  1`] = `
<div
  className="HeaderAccountMenu varClassName"
>
  <button
    aria-controls={null}
    aria-expanded={false}
    aria-haspopup="true"
    className="HeaderAccountMenu__button"
    id="accountMenuBtn"
    onClick={[Function]}
    onKeyDown={[Function]}
    type="button"
  >
    <HeaderIcon
      icon={
        Object {
          "icon": Array [
            448,
            512,
            Array [
              128100,
              62144,
            ],
            "f007",
            "M224 256A128 128 0 1 0 224 0a128 128 0 1 0 0 256zm-45.7 48C79.8 304 0 383.8 0 482.3C0 498.7 13.3 512 29.7 512H418.3c16.4 0 29.7-13.3 29.7-29.7C448 383.8 368.2 304 269.7 304H178.3z",
          ],
          "iconName": "user",
          "prefix": "fas",
        }
      }
      label="Account"
    />
  </button>
</div>
`;

exports[`HeaderAccountMenu Rendering Render HeaderAccountMenu component with pseudo translated text 1`] = `
<div
  className="HeaderAccountMenu"
>
  <button
    aria-controls={null}
    aria-expanded={false}
    aria-haspopup="true"
    className="HeaderAccountMenu__button"
    id="accountMenuBtn"
    onClick={[Function]}
    onKeyDown={[Function]}
    type="button"
  >
    <HeaderIcon
      icon={
        Object {
          "icon": Array [
            448,
            512,
            Array [
              128100,
              62144,
            ],
            "f007",
            "M224 256A128 128 0 1 0 224 0a128 128 0 1 0 0 256zm-45.7 48C79.8 304 0 383.8 0 482.3C0 498.7 13.3 512 29.7 512H418.3c16.4 0 29.7-13.3 29.7-29.7C448 383.8 368.2 304 269.7 304H178.3z",
          ],
          "iconName": "user",
          "prefix": "fas",
        }
      }
      label="ÂÂççççôôùùnt"
    />
  </button>
</div>
`;
